DToCToDB.md

```diff
@@ -0,0 +1,1193 @@
+mar_16 = pd.read_csv('https://www.england.nhs.uk/statistics/wp-content/uploads/sites/2/2015/05/Monthly-SITREPs-DTOC-Extracts-iFYJ2.csv', header=3)
+feb_16 = pd.read_csv('https://www.england.nhs.uk/statistics/wp-content/uploads/sites/2/2015/05/Monthly-SITREPs-DTOC-Extracts-gVela.csv', header=3)
+jan_16 = pd.read_csv('https://www.england.nhs.uk/statistics/wp-content/uploads/sites/2/2015/05/Monthly-SITREPs-DTOC-Extracts-9jMj2.csv', header=3)
+```
+
+We then combine the 2016 dataframes in a single dataframe assigned to `dtoc_16` using the `pd.concat()` function:
+
+
+```python
+frames = [oct_16, sep_16, jul_16, jun_16, may_16, apr_16, mar_16, feb_16, jan_16]
+
+dtoc_16 = pd.concat(frames)
+dtoc_16.info()
+```
+
+    <class 'pandas.core.frame.DataFrame'>
+    Int64Index: 172000 entries, 0 to 18459
+    Data columns (total 16 columns):
+    Year                              172000 non-null object
+    Period Name                       172000 non-null object
+    Provider Parent Org Code          172000 non-null object
+    Provider Parent Name              172000 non-null object
+    Provider Org Code                 172000 non-null object
+    Provider Org Name                 172000 non-null object
+    Local Authority Code              172000 non-null object
+    Local Authority Name              172000 non-null object
+    Acute Or Non Acute Description    172000 non-null object
+    Reason For Delay                  172000 non-null object
+    NHS A SUM                         171996 non-null float64
+    NHS B SUM                         171995 non-null float64
+    Social Care A SUM                 172000 non-null int64
+    Social Care B SUM                 171998 non-null float64
+    Both A SUM                        172000 non-null int64
+    Both B SUM                        171998 non-null float64
+    dtypes: float64(4), int64(2), object(10)
+    memory usage: 22.3+ MB
+
+
+Note the null values in the dataframe, for now we will ignore this and deal with it once we have combinded all the datasets.
+
+Currently the date information is stored as strings in two different columns, with the year in the financial year format and the month as a capitalised name. It will be helpful once we have combinded all the dataframes to add a true datetime column, therefore we will do some preperation work by modifying these columns.
+
+* Year
+  * Convert to a integer value for the calender year.
+* Month
+  * Change the name of the 'Period Name' column to 'Month'
+  * Convert the capatilised month names to an integer value.
+
+
+```python
+# Year
+dtoc_16['Year'] = 2016
+
+# Month
+dtoc_16 = dtoc_16.rename(columns={'Period Name':'Month'})
+date_str = ['JANUARY','FEBRUARY','MARCH','APRIL','MAY','JUNE','JULY','AUGUST','SEPTEMBER','OCTOBER','NOVEMBER','DECEMBER']
+date_int = [1,2,3,4,5,6,7,8,9,10,11,12]
+for i in range(0,12):
+    dtoc_16['Month'] = dtoc_16['Month'].replace(date_str[i],date_int[i])    
```

Comment:
You can simplify this logic with [pandas.Series.map](http://pandas.pydata.org/pandas-docs/stable/generated/pandas.Series.map.html).  ie

```
dtoc_16['Month'].map(lambda x: date_str.index(x) + 1)
```

Then you don't even need the for loop, etc.